src/btree/col_srch.c

```diff
@@ -134,6 +140,10 @@ __wt_col_search(
         base = pindex->entries;
         descent = pindex->index[base - 1];
 
+        if (btree->type == BTREE_COL_FIX && !greater_recno_exists_flcs &&
+          descent->ref_recno > recno)
+            greater_recno_exists_flcs = true;
```

Comment:
Can we remove this local variable as well?

Reply:
We need to save the information that we have seen a greater record, I cannot do that check at the end. Are you thinking of a different way to handle this?

Reply:
I have the same understanding just want to check. May be add a comment here to explain we need to check this here.

Reply:
Sure, sounds good. Added one in b4f37473b4be321d3cc296a9f291c129810c5659.